src/RFC-0314-VNCSelection.md

```diff
@@ -0,0 +1,88 @@
+"NOT RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in 
+[BCP 14](https://tools.ietf.org/html/bcp14) (covering RFC2119 and RFC8174) when, and only when, they appear in all capitals, as 
+shown here.
+
+## Disclaimer
+
+This document and its content are intended for information purposes only and may be subject to change or update
+without notice.
+
+This document may include preliminary concepts that may or may not be in the process of being developed by the Tari
+community. The release of this document is intended solely for review and discussion by the community of the
+technological merits of the potential system outlined herein.
+
+## Goals
+The goal of this RFC is to describe the process for allocating Validator Nodes (VNs) to Validator Node Committees (VNCs)
+
+## Intro
+Validator nodes will have to group themselves into Validator Node Committees for transactions/shard processing. Committees will be formed for each new transaction/shard processing. This committee needs to be determined randomly, and we use the VN_Key as this changes periodically and is random. Each VN will be allocated a unique and individual shard space to serve in as a VNC member. 
+
+## Committee Creation
+Because we have the base layer where each VN needs to publish a registration transaction, we can get base_node and miners to keep track of all active VNs. We represent all VNs in a (balanced) Merkle tree with the VN_keys as leaves. We declare a constant `COMMITTEE_SIZE` which can be changed in the consensus constants.
+When a transaction is proposed, the shards for each substate determine the VN committee from the Merkle tree by finding the `COMMITTEE_SIZE / 2` VN_keys to the 
+left and `COMMITTEE_SIZE / 2` VN_Keys to the right in the Merkle tree.
+
+This makes it very easy to determine what shard space a VN needs to serve, which states the VN needs to sync from peers, and who has them. Because the
+second layer has a delayed view of the network. VNs can also know beforehand and prep to ensure they are ready when new block heights appear. 
+
+## Committee proofs
+We construct the balanced Merkle tree from all active VN registration transactions and prune away all inactive ones. Because the [base node]s have to keep track
+of the entire unspent UTXO set, it becomes easy for them to track and validate all active VN registration UTXOs. This means we can keep [base node]s 
+responsible for validating and constructing the Merkle tree. We commit this Merkle tree per block as a Merkle root inside the block's header as the `validator_node_set_root`. 
+
+This Merkle root in the header always lags by one block, meaning that the Merkle root is for the state of the VN's before the start of the block it's mined in. 
+When a VN registers, that new VN key will only appear inside the Merkle tree in the next block. The reason for this is that header_hash is used in the
+the calculation process of the VN_key. 

```

Comment:
This needs some more background and justification.

There are a few goals that base node registration should be looking to achieve:
1. Sybil resistance
2. Providing the source of truth for the population of VNs at all times.
3. Providing the basis for the mapping of VNs to shard space
4. Managing VN registration and de-registration
5. Providing a level of stability regarding the validator node population

1, 2, 4 can be fully accomplished by using a new type of output feature marking an output as a `VN_REGISTRATION` transaction. The UTXO will lock up some non-negligible amount (`VN_DEPOSIT` of say 10,000 XTR) for a minimum amount of time (`VN_DEPOSIT_PERIOD`), say 3 months. 

The deposit and time lock act as an effective form of Sybil resistance (Goal 1). 
The set of inspect `VN_REGISTRATION` outputs represents the population of validator nodes (Goal 2).
After the minimum time period, VN operators can spend the `VN_REGISTRATION` output to deregister the node (Goal 4). 

A more aggressive alternative is to _require_ via an output script & covenant that the UTXO gets renewed or deregistered every 3 months, otherwise, the UTXO is spendable by anyone to a burn output.

Goal 3 is partially achieved via the registration transaction. However, we must shuffle VN's shard groups periodically to prevent strategies by node operators to carry out some [shard targeting attacks]*. This is done by introducing the concept of an _epoch_ into the VN population management philosophy. 

 Epochs are periods of time, of roughly hours (720 blocks), that commit to, and define the current VN set. Any transactions involving `VN_REGISTRATION` outputs do not take effect until the start of the next epoch. This lets us achieve Goal 5 directly, and provides the additional security to achieve Goal 3. There are some other nuances of epoch management that require their own RFC, but broadly speaking, the base layer PoW chain acts as a big, slow metronome keeping time on the DAN.

*explanation required

Reply:
Busy working on RFC_0313 which deals with VN Registration. This will address most of this. 
Do you think its perhaps better to merge the two, rather than have one for registration and one for VNC selection?